URIUtilTest.testFileUriGetUriLastPathSegment fails on recent macOS

**Jetty version(s)**
12.0.5-SNAPSHOT

**Jetty Environment**
any

**Java version/vendor** `(use: java -version)`
any

**OS type/version**
macOS 14.2

**Description**
```
[ERROR] Failures: 
[ERROR]   URIUtilTest.testFileUriGetUriLastPathSegment:818 
Expected: is "número.war"
     but: was "número.war"
[ERROR]   UrlResourceFactoryTest.testHttps:67 expected: <true> but was: <false>
```

The test code has a special case for macOS around NFD form normalization. Apparently that is no longer the case for macOS, at least on 14.2.

**How to reproduce?**
```
cd jetty-core/jetty-util ; mvn clean install
```

--- a/jetty-core/jetty-util/src/test/java/org/eclipse/jetty/util/URIUtilTest.java
+++ b/jetty-core/jetty-util/src/test/java/org/eclipse/jetty/util/URIUtilTest.java
@@ -810,12 +810,24 @@ public class URIUtilTest
             FS.touch(base);
         }
         URI uri = base.toUri();
-        if (OS.MAC.isCurrentOs())
+
+        try
         {
-            // Normalize Unicode to NFD form that OSX Path/FileSystem produces
-            expectedName = Normalizer.normalize(expectedName, Normalizer.Form.NFD);
+            assertThat(URIUtil.getUriLastPathSegment(uri), is(expectedName));
         }
-        assertThat(URIUtil.getUriLastPathSegment(uri), is(expectedName));
+        catch (AssertionError e)
+        {
+            if (OS.MAC.isCurrentOs())
+            {
+                // Normalize Unicode to NFD form that OSX Path/FileSystem produces
+                expectedName = Normalizer.normalize(expectedName, Normalizer.Form.NFD);
+                assertThat(URIUtil.getUriLastPathSegment(uri), is(expectedName));
+            }
+            else
+            {
+                throw e;
+            }
+        }
     }
 
     public static Stream<Arguments> uriLastSegmentSource() throws IOException

